www/include/configuration/configObject/service/DB-Func.php

```diff
@@ -660,21 +721,42 @@ function multipleServiceInDB(
                             "servicegroup_relation WHERE service_service_id = '" . $key . "'";
                         $dbResult = $pearDB->query($query);
                         $fields["service_sgs"] = "";
+                        $query = "INSERT INTO servicegroup_relation (host_host_id, hostgroup_hg_id, " .
+                                 "service_service_id, servicegroup_sg_id)
+                                 VALUES (:host_host_id,:hostgroup_hg_id,:service_service_id,:servicegroup_sg_id)";
+                        $statement = $pearDB->prepare($query);
                         while ($Sg = $dbResult->fetch()) {
                             if (isset($host) && $host) {
                                 $host_id = $host;
                             } else {
-                                $Sg["host_host_id"] ? $host_id = "'" . $Sg["host_host_id"] . "'" : $host_id = "NULL";
+                                $host_id = $Sg["host_host_id"] ?? null;
                             }
                             if (isset($hostgroup) && $hostgroup) {
                                 $hg_id = $hostgroup;
                             } else {
-                                $Sg["hostgroup_hg_id"] ? $hg_id = "'" . $Sg["hostgroup_hg_id"] . "'" : $hg_id = "NULL";
+                                $hg_id = $Sg["hostgroup_hg_id"] ?? null;
                             }
-                            $query = "INSERT INTO servicegroup_relation (host_host_id, hostgroup_hg_id, " .
-                                "service_service_id, servicegroup_sg_id) VALUES (" . $host_id . ", " . $hg_id . ", '" .
-                                $maxId["MAX(service_id)"] . "', '" . $Sg["servicegroup_sg_id"] . "')";
-                            $pearDB->query($query);
+                            $statement->bindValue(
+                                ':host_host_id',
+                                (int) $host_id,
+                                $host_id ? \PDO::PARAM_INT : \PDO::PARAM_NULL
```

Comment:
The issue is that you always cast $host_id, so if $host_id is null, you will insert `0`
```suggestion
                                $host_id,
                                \PDO::PARAM_INT
```